src/components/table/column-picker/column-picker.component.ts

```diff
@@ -41,39 +70,163 @@ export class ColumnPickerComponent {
     /** Define a function that return a column move announcement */
     @Input() columnMovedAnnouncement: (column: string, delta: number) => string = this.getColumnMovedAnnouncement;
 
+    /** group settings for customising display */
+    @Input() groupSettings: ColumnPickerGroupSetting[] = [];
+
     /** Emits when the selected items change */
-    @Output() selectedChange = new EventEmitter<ReadonlyArray<string>>();
+    @Output() selectedChange = new EventEmitter<ReadonlyArray<ColumnPickerValue>>();
 
     /** Emits when the deselected items change */
-    @Output() deselectedChange = new EventEmitter<ReadonlyArray<string>>();
+    @Output() deselectedChange = new EventEmitter<ReadonlyArray<ColumnPickerValue>>();
+
+    /** The Nested tree control used for the deselect tree */
+    treeControl: NestedTreeControl<ColumnPickerTreeNode> = new NestedTreeControl<ColumnPickerTreeNode>(item => item.children);
+
+    /** A tree-friendly representation of the deselected data */
+    treeData: ColumnPickerTreeNode[];
+
+    /** Data source observable bound to the tree control */
+    treeDataSource: ArrayDataSource<ColumnPickerTreeNode>;
 
     /** Store the list of deselected columns that can be moved */
-    _deselectedSelection: ReadonlyArray<string> = [];
+    _deselectedSelection: ReadonlyArray<ColumnPickerValue> = [];
 
     /** Store the list of selected columns that can be moved */
-    _selectedSelection: ReadonlyArray<string> = [];
+    _selectedSelection: ReadonlyArray<ColumnPickerValue> = [];
+
+    /** Defines the index of the item currently in focus */
+    activeIndex: number = 0;
+
+    isFocused: boolean = false;
 
     /** Cache selection during reordering */
-    private _selection: ReadonlyArray<string> = [];
+    private _selection: ReadonlyArray<ColumnPickerValue> = [];
+
+    /** Focus manager for cycling through the tree nodes */
+    private _focusKeyManager: FocusKeyManager<CdkNestedTreeNode<ColumnPickerTreeNode>>;
```

Comment:
This needs work. I'm having some issues with getting the keyboard events to trigger on all items and cycle through the heirarchy - it tends to only work on the currently focused level, so subnodes of a group or top level nodes only.